generator/migration.go

```diff
@@ -0,0 +1,670 @@
+		return JSONBColumn, nil
+	}
+
+	if f.Kind == Array || f.Kind == Slice {
+		return ArrayColumn(typeMappings[removeTypePrefix(f.Type)]), nil
+	}
+
+	if pk {
+		if !isValidIdentifier(f) {
+			return ColumnType(""), fmt.Errorf("kallax: type %s is not a valid type for a primary key. On field %s of model %s.", f.Type, f.Name, f.Model.Name)
+		}
+
+		return idTypeMappings[identifierType(f)], nil
+	}
+
+	if f.Kind == Basic {
+		typ, ok := typeMappings[f.Type]
+		if !ok {
+			return ColumnType(""), fmt.Errorf("kallax: type %s can not be converted to a SQL type. On field %s of model %s. Consider using the struct tag `sqltype` to set a custom type for this column.", f.Type, f.Name, f.Model.Name)
+		}
+		return typ, nil
+	}
+
+	if f.Kind == Relationship && !f.IsInverse() {
+		typ := removeTypePrefix(f.Type)
+		table, ok := t.tableIndex[typ]
+		if !ok {
+			return ColumnType(""), fmt.Errorf("kallax: unable to find table for type %s in field %s of model %s. Is the model type part of the generation input?", typ, f.Name, f.Model.Name)
+		}
+
+		return t.transformType(t.pkIndex[table], false)
+	}
+
+	if f.Kind == Relationship {
+		return t.transformType(f.Model.ID, false)
+	}
+
+	if f.Kind == Interface {
+		typ := removeTypePrefix(typeName(f.Node.Type()))
+		if typ, ok := typeMappings[typ]; ok {
+			return typ, nil
+		}
+	}
+
+	return ColumnType(""), fmt.Errorf("kallax: cannot find a suitable type (%s) for field %s of model %s. Consider using the struct tag `sqltype` to set a custom type for this column.", f.Type, f.Name, f.Model.Name)
+}
+
+func (t *packageTransformer) transformRef(f *Field) (*Reference, error) {
+	if f.Kind == Relationship && !f.IsInverse() {
+		typ := removeTypePrefix(f.Type)
+		table, ok := t.tableIndex[typ]
+		if !ok {
+			return nil, fmt.Errorf("kallax: unable to find table for type %s in field %s of model %s. Is the model type part of the generation input?", typ, f.Name, f.Model.Name)
+		}
+
+		return &Reference{Table: table, Column: t.pkIndex[table].ColumnName()}, nil
+	} else if f.Kind == Relationship {
+		return &Reference{Table: f.Model.Table, Column: f.Model.ID.ColumnName()}, nil
+	}
+
+	return nil, nil
+}
+
+var typeMappings = map[string]ColumnType{
+	"gopkg.in/src-d/go-kallax.v1.ULID":      UUIDColumn,
+	"gopkg.in/src-d/go-kallax.v1.UUID":      UUIDColumn,
+	"gopkg.in/src-d/go-kallax.v1.NumericID": BigIntColumn,
+	"github.com/satori/go.uuid.UUID":        UUIDColumn,
+	"string":        TextColumn,
+	"rune":          TextColumn,
+	"uint8":         SmallIntColumn,
+	"int8":          SmallIntColumn,
+	"byte":          SmallIntColumn,
+	"uint16":        IntegerColumn,
+	"int16":         SmallIntColumn,
+	"uint32":        BigIntColumn,
+	"int32":         IntegerColumn,
+	"uint":          NumericColumn(20),
+	"int":           BigIntColumn,
+	"int64":         BigIntColumn,
+	"uint64":        NumericColumn(20),
+	"float32":       RealColumn,
+	"float64":       DoubleColumn,
+	"bool":          BooleanColumn,
+	"url.URL":       TextColumn,
```

Comment:
we will need a way to set size of a string field for use `TextColumn` or `BarcharColumn(n)`